backend/uclapi/oauth/views.py

```diff
@@ -228,9 +226,46 @@ def shibcallback(request):
     }
 
     initial_data = json.dumps(page_data, cls=DjangoJSONEncoder)
-    return render(request, 'permissions.html', {
-        'initial_data': initial_data
-    })
+
+    try:
+        token = OAuthToken.objects.get(app=app, user=user)
```

Comment:
AFAICT this is the only line that can a `OAuthToken.DoesNotExist` exception, so the try/catch block should only surround this line.

Reply:
That makes a lot of sense, should probably do that in the rest of the file too retroactively at some point.